lib/Sema/CSSimplify.cpp

```diff
@@ -2793,6 +2793,26 @@ bool ConstraintSystem::repairFailures(
   return !conversionsOrFixes.empty();
 }
 
+static bool
+shouldDiagnoseUnnecessaryExplicitCoercion(ConstraintSystem &CS,
+                                          ConstraintKind kind,
+                                          ConstraintLocatorBuilder locator) {
```

Comment:
I think this logic should be associated with the fix directly, I have been using `attempt` as a method name

Reply:
Sure, done